tools/cldr-apps/js/src/views/DashboardWidget.vue

```diff
@@ -367,22 +375,47 @@ export default {
     },
 
     catCheckmarkChanged(event, category) {
-      // setTimeout solves a weakness in the Vue implementation: if the number of
-      // notifications is large, the checkbox in the header can take a second or more
+      // setTimeout is intended to solve a weakness in the Vue implementation: if the number of
+      // notifications is large, the checkbox in the header can take a second or even a minute
       // to change its visible state in response to the user's click, during which time
       // the user may click again thinking the first click wasn't recognized. Postponing
-      // the DOM update of thousands of rows ensures that the header checkbox updates
+      // the DOM update of thousands of rows should help ensure that the header checkbox updates
       // without delay.
-      this.updatingVisibility = true;
-      setTimeout(
-        () => this.updateVisibility(event.target.checked, category),
-        0
+      // Also the booleans catCheckboxIsUnchecked and catIsHidden are distinct in order for
+      // the checkbox itself to update immediately even if the rows for the corresponding
+      // category may take a long time to update.
+      // Unfortunately, neither of these mechanisms seems guaranteed to prevent a very very
+      // long delay between the time the user clicks the checkbox and the time that the checkbox
+      // changes its state.
```

Comment:
I wouldn't say it's a weakness in Vue itself but a weakness in our usage of the order of 10k DOM objects.